### PR TITLE
Adding Hugepage as resource type

### PR DESCRIPTION
**Pull request by fawaz-moh**
Adding resource type hugepages- as a valid resource can be requested. Hugepage is GA as of version 1.16. more info can be found [here](https://kubernetes.io/docs/tasks/manage-hugepages/scheduling-hugepages/)

### CODE REVIEW

**Comment by zacharysarah** [comment]
@fawaz-moh Can you please update the description to indicate what you're fixing and why? Thanks in advance!

File: content/en/docs/concepts/configuration/manage-compute-resources-container.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ the difference between requests and limits, see
 
 *CPU* and *memory* are each a *resource type*. A resource type has a base unit.
 CPU is specified in units of cores, and memory is specified in units of bytes.
+Starting from version 1.16, *Hugepages* also can be specified. Typical sizes are 2Mi and 1Gi.
+Unlike CPU and memory, Hugepages do not support overcommit.
```
Comment (sftim) [inline content/en/docs/concepts/configuration/manage-compute-resources-container.md]:
```suggestion
{{< note >}}
You cannot overcommit `hugepages-*` resources.
This is different from the `memory` and `cpu` resources.
{{< /note >}}
```

File: content/en/docs/concepts/configuration/manage-compute-resources-container.md [hunk 2]
```diff
@@ -27,6 +27,8 @@ the difference between requests and limits, see
 
 *CPU* and *memory* are each a *resource type*. A resource type has a base unit.
 CPU is specified in units of cores, and memory is specified in units of bytes.
+Starting from version 1.16, *Hugepages* also can be specified. Typical sizes are 2Mi and 1Gi.
```
Comment (sftim) [inline content/en/docs/concepts/configuration/manage-compute-resources-container.md]: I'm not sure what this is saying to me.

How about:
```markdown
If you're using Kubernetes v1.14 or newer, you can specify _huge page_ resources.
Huge pages are a Linux-specific feature where the node kernel allocates blocks of memory
that are much larger than the default page size.

For example, on a system where the default page size is 4KiB, you could specify a limit,
`hugepages-2Mi: 80Mi`. If the container tries allocating over 40 2MiB huge pages (a
total of 80 MiB) in total, that allocation fails.
```

(according to https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ HugePages went GA in v1.14 - if that's wrong, there more fixing needed).

Comment (fawaz-moh) [reply content/en/docs/concepts/configuration/manage-compute-resources-container.md]: The suggested changes make more sense @sftim

File: content/en/docs/concepts/configuration/manage-compute-resources-container.md [hunk 3]
```diff
@@ -27,8 +27,18 @@ the difference between requests and limits, see
...
 CPU is specified in units of cores, and memory is specified in units of bytes.
-Starting from version 1.16, *Hugepages* also can be specified. Typical sizes are 2Mi and 1Gi.
-Unlike CPU and memory, Hugepages do not support overcommit.
+If you're using Kubernetes v1.14 or newer, you can specify _huge page_ resources.
+Huge pages are a Linux-specific feature where the node kernel allocates blocks of memory
+that are much larger than the default page size.
+
+For example, on a system where the default page size is 4KiB, you could specify a limit,
+`hugepages-2Mi: 80Mi`. If the container tries allocating over 40 2MiB huge pages (a
+total of 80 MiB) in total, that allocation fails.
```
Comment (sftim) [inline content/en/docs/concepts/configuration/manage-compute-resources-container.md]: Oops, repeated _total_ there!

Comment (fawaz-moh) [reply content/en/docs/concepts/configuration/manage-compute-resources-container.md]: Updated

**Pull request merged**